2025-06-06 14:00:01 - Creating client for connecting using MQTT over TCP
2025-06-06 14:00:01 - Connecting to Hub...
2025-06-06 14:00:01 - Connect using port 8883 (TCP)
2025-06-06 14:00:02 - connected with result code: 0
2025-06-06 14:00:02 - _on_mqtt_connected called
2025-06-06 14:00:02 - Connection State - Connected
2025-06-06 14:00:02 - Successfully connected to Hub
2025-06-06 14:00:02 - Enabling feature:methods...
2025-06-06 14:00:02 - subscribing to $iothub/methods/POST/# with qos 1
2025-06-06 14:00:03 - suback received for 1
2025-06-06 14:00:03 - Successfully enabled feature:methods
2025-06-06 14:00:03 - Khởi động hệ thống
2025-06-06 14:07:24 - message received on $iothub/methods/POST/Agitator/?$rid=1
2025-06-06 14:07:24 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:07:24 - ['azure.iot.device.iothub.sync_handler_manager.HandlerManagerException: HANDLER (_on_method_request_received): Error during invocation\n']
2025-06-06 14:09:24 - message received on $iothub/methods/POST/Agitator/?$rid=2
2025-06-06 14:09:24 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:09:24 - ['azure.iot.device.iothub.sync_handler_manager.HandlerManagerException: HANDLER (_on_method_request_received): Error during invocation\n']
2025-06-06 14:13:01 - message received on $iothub/methods/POST/Agitator/?$rid=3
2025-06-06 14:13:01 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:13:01 - ['azure.iot.device.iothub.sync_handler_manager.HandlerManagerException: HANDLER (_on_method_request_received): Error during invocation\n']
2025-06-06 14:14:15 - Creating client for connecting using MQTT over TCP
2025-06-06 14:14:15 - Connecting to Hub...
2025-06-06 14:14:15 - Connect using port 8883 (TCP)
2025-06-06 14:14:28 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:14:28 - transport.connect raised error
2025-06-06 14:14:28 - Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 396, in connect
    rc = self._mqtt_client.connect(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 914, in connect
    return self.reconnect()
           ^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 1044, in reconnect
    sock = self._create_socket_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 3685, in _create_socket_connection
    return socket.create_connection(addr, timeout=self._connect_timeout, source_address=source)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 851, in create_connection
    raise exceptions[0]
  File "/usr/lib/python3.11/socket.py", line 836, in create_connection
    sock.connect(sa)
TimeoutError: timed out

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py", line 206, in _run_op
    self.transport.connect(password=password)
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 419, in connect
    raise exceptions.ConnectionFailedError() from e
azure.iot.device.common.transport_exceptions.ConnectionFailedError

2025-06-06 14:14:28 - Callback completed with error 
2025-06-06 14:14:28 - NoneType: None

2025-06-06 14:15:33 - Creating client for connecting using MQTT over TCP
2025-06-06 14:15:33 - Connecting to Hub...
2025-06-06 14:15:33 - Connect using port 8883 (TCP)
2025-06-06 14:15:33 - connected with result code: 0
2025-06-06 14:15:33 - _on_mqtt_connected called
2025-06-06 14:15:33 - Connection State - Connected
2025-06-06 14:15:33 - Successfully connected to Hub
2025-06-06 14:15:33 - Enabling feature:methods...
2025-06-06 14:15:33 - subscribing to $iothub/methods/POST/# with qos 1
2025-06-06 14:15:33 - disconnected with result code: 7
2025-06-06 14:15:33 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:15:33 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:15:33 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:15:33 - Connection State - Disconnected
2025-06-06 14:15:33 - Cleared all pending method requests due to disconnect
2025-06-06 14:15:33 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:15:33 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:15:33 - Connect using port 8883 (TCP)
2025-06-06 14:15:34 - suback received for 1
2025-06-06 14:15:34 - Successfully enabled feature:methods
2025-06-06 14:15:34 - Khởi động hệ thống
2025-06-06 14:15:35 - connected with result code: 0
2025-06-06 14:15:35 - _on_mqtt_connected called
2025-06-06 14:15:35 - Connection State - Connected
2025-06-06 14:15:35 - disconnected with result code: 7
2025-06-06 14:15:35 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:15:35 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:15:35 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:15:35 - Connection State - Disconnected
2025-06-06 14:15:35 - Cleared all pending method requests due to disconnect
2025-06-06 14:15:35 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:15:35 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:15:35 - Connect using port 8883 (TCP)
2025-06-06 14:15:39 - connected with result code: 0
2025-06-06 14:15:39 - disconnected with result code: 7
2025-06-06 14:15:39 - _on_mqtt_connected called
2025-06-06 14:15:39 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:15:39 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:15:39 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:15:39 - Connection State - Connected
2025-06-06 14:15:39 - Connection State - Disconnected
2025-06-06 14:15:39 - Cleared all pending method requests due to disconnect
2025-06-06 14:15:39 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:15:39 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:15:39 - Connect using port 8883 (TCP)
2025-06-06 14:15:40 - message received on $iothub/methods/POST/Agitator/?$rid=1
2025-06-06 14:15:40 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:15:40 - ['azure.iot.device.iothub.sync_handler_manager.HandlerManagerException: HANDLER (_on_method_request_received): Error during invocation\n']
2025-06-06 14:15:43 - connected with result code: 0
2025-06-06 14:15:43 - _on_mqtt_connected called
2025-06-06 14:15:43 - disconnected with result code: 7
2025-06-06 14:15:43 - Connection State - Connected
2025-06-06 14:15:43 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:15:43 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:15:43 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:15:43 - Connection State - Disconnected
2025-06-06 14:15:43 - Cleared all pending method requests due to disconnect
2025-06-06 14:15:43 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:15:43 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:15:43 - Connect using port 8883 (TCP)
2025-06-06 14:15:54 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:15:54 - transport.connect raised error
2025-06-06 14:15:54 - Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 396, in connect
    rc = self._mqtt_client.connect(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 914, in connect
    return self.reconnect()
           ^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 1044, in reconnect
    sock = self._create_socket_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 3685, in _create_socket_connection
    return socket.create_connection(addr, timeout=self._connect_timeout, source_address=source)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 851, in create_connection
    raise exceptions[0]
  File "/usr/lib/python3.11/socket.py", line 836, in create_connection
    sock.connect(sa)
TimeoutError: timed out

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py", line 206, in _run_op
    self.transport.connect(password=password)
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 419, in connect
    raise exceptions.ConnectionFailedError() from e
azure.iot.device.common.transport_exceptions.ConnectionFailedError

2025-06-06 14:15:54 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:15:54 - ['azure.iot.device.common.transport_exceptions.ConnectionFailedError\n']
2025-06-06 14:15:56 - message received on $iothub/methods/POST/Agitator/?$rid=1
2025-06-06 14:15:56 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:15:56 - ['azure.iot.device.iothub.sync_handler_manager.HandlerManagerException: HANDLER (_on_method_request_received): Error during invocation\n']
2025-06-06 14:16:04 - Connect using port 8883 (TCP)
2025-06-06 14:16:14 - connected with result code: 0
2025-06-06 14:16:14 - _on_mqtt_connected called
2025-06-06 14:16:14 - Connection State - Connected
2025-06-06 14:16:14 - disconnected with result code: 7
2025-06-06 14:16:14 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:16:14 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:16:14 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:16:14 - Connection State - Disconnected
2025-06-06 14:16:14 - Cleared all pending method requests due to disconnect
2025-06-06 14:16:14 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:16:14 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:16:14 - Connect using port 8883 (TCP)
2025-06-06 14:16:17 - message received on $iothub/methods/POST/RunAll/?$rid=1
2025-06-06 14:16:17 - connected with result code: 0
2025-06-06 14:16:17 - _on_mqtt_connected called
2025-06-06 14:16:17 - Connection State - Connected
2025-06-06 14:16:24 - disconnected with result code: 7
2025-06-06 14:16:24 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:16:24 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:16:24 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:16:24 - Connection State - Disconnected
2025-06-06 14:16:24 - Cleared all pending method requests due to disconnect
2025-06-06 14:16:24 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:16:24 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:16:24 - Connect using port 8883 (TCP)
2025-06-06 14:16:29 - Bơm 1 bắt đầu chạy liên tục
2025-06-06 14:16:29 - Bơm 2 bắt đầu chạy liên tục
2025-06-06 14:16:29 - Bơm 3 bắt đầu chạy liên tục
2025-06-06 14:16:29 - Bơm 4 bắt đầu chạy liên tục
2025-06-06 14:16:29 - Bơm 5 bắt đầu chạy liên tục
2025-06-06 14:16:29 - Bơm 6 bắt đầu chạy liên tục
2025-06-06 14:16:29 - Bơm 7 bắt đầu chạy liên tục
2025-06-06 14:16:29 - Bơm 8 bắt đầu chạy liên tục
2025-06-06 14:16:29 - Tất cả các bơm đang chạy liên tục
2025-06-06 14:16:29 - Bơm 9 bắt đầu chạy liên tục
2025-06-06 14:16:30 - Sending method response to Hub...
2025-06-06 14:16:43 - message received on $iothub/methods/POST/RunAll/?$rid=1
2025-06-06 14:16:43 - Bơm 1 bắt đầu chạy liên tục
2025-06-06 14:16:43 - Bơm 2 bắt đầu chạy liên tục
2025-06-06 14:16:43 - Bơm 3 bắt đầu chạy liên tục
2025-06-06 14:16:43 - Bơm 4 bắt đầu chạy liên tục
2025-06-06 14:16:43 - Bơm 5 bắt đầu chạy liên tục
2025-06-06 14:16:43 - Bơm 6 bắt đầu chạy liên tục
2025-06-06 14:16:43 - Bơm 7 bắt đầu chạy liên tục
2025-06-06 14:16:43 - Bơm 8 bắt đầu chạy liên tục
2025-06-06 14:16:43 - Bơm 9 bắt đầu chạy liên tục
2025-06-06 14:16:43 - Tất cả các bơm đang chạy liên tục
2025-06-06 14:16:44 - Sending method response to Hub...
2025-06-06 14:16:44 - publishing on $iothub/methods/res/200/?$rid=1
2025-06-06 14:16:44 - payload published for 2
2025-06-06 14:16:44 - Successfully sent method response to Hub
2025-06-06 14:16:59 - disconnected with result code: 7
2025-06-06 14:16:59 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:16:59 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:16:59 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:16:59 - Connection State - Disconnected
2025-06-06 14:16:59 - Cleared all pending method requests due to disconnect
2025-06-06 14:16:59 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:16:59 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:16:59 - Connect using port 8883 (TCP)
2025-06-06 14:16:59 - connected with result code: 0
2025-06-06 14:16:59 - _on_mqtt_connected called
2025-06-06 14:16:59 - publishing on $iothub/methods/res/200/?$rid=1
2025-06-06 14:16:59 - Connection State - Connected
2025-06-06 14:16:59 - payload published for 2
2025-06-06 14:16:59 - Successfully sent method response to Hub
2025-06-06 14:17:06 - connected with result code: 0
2025-06-06 14:17:06 - _on_mqtt_connected called
2025-06-06 14:17:06 - disconnected with result code: 7
2025-06-06 14:17:06 - Connection State - Connected
2025-06-06 14:17:06 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:17:06 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:17:06 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:17:06 - Connection State - Disconnected
2025-06-06 14:17:06 - Cleared all pending method requests due to disconnect
2025-06-06 14:17:06 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:06 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:17:06 - Connect using port 8883 (TCP)
2025-06-06 14:17:06 - message received on $iothub/methods/POST/Agitator/?$rid=1
2025-06-06 14:17:06 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:06 - ['azure.iot.device.iothub.sync_handler_manager.HandlerManagerException: HANDLER (_on_method_request_received): Error during invocation\n']
2025-06-06 14:17:08 - disconnected with result code: 7
2025-06-06 14:17:08 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:17:08 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:17:08 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:17:08 - Connection State - Disconnected
2025-06-06 14:17:08 - Cleared all pending method requests due to disconnect
2025-06-06 14:17:08 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:08 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:17:08 - Connect using port 8883 (TCP)
2025-06-06 14:17:08 - connected with result code: 0
2025-06-06 14:17:08 - _on_mqtt_connected called
2025-06-06 14:17:08 - Connection State - Connected
2025-06-06 14:17:26 - connected with result code: 0
2025-06-06 14:17:26 - _on_mqtt_connected called
2025-06-06 14:17:26 - Connection State - Connected
2025-06-06 14:17:26 - disconnected with result code: 7
2025-06-06 14:17:26 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:17:26 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:17:26 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:17:26 - Connection State - Disconnected
2025-06-06 14:17:26 - Cleared all pending method requests due to disconnect
2025-06-06 14:17:26 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:26 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:17:26 - Connect using port 8883 (TCP)
2025-06-06 14:17:28 - connected with result code: 0
2025-06-06 14:17:28 - _on_mqtt_connected called
2025-06-06 14:17:28 - Connection State - Connected
2025-06-06 14:17:28 - disconnected with result code: 7
2025-06-06 14:17:28 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:17:28 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:17:28 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:17:28 - Connection State - Disconnected
2025-06-06 14:17:28 - Cleared all pending method requests due to disconnect
2025-06-06 14:17:28 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:28 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:17:28 - Connect using port 8883 (TCP)
2025-06-06 14:17:29 - connected with result code: 0
2025-06-06 14:17:29 - _on_mqtt_connected called
2025-06-06 14:17:29 - Connection State - Connected
2025-06-06 14:17:29 - disconnected with result code: 7
2025-06-06 14:17:29 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:17:29 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:17:29 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:17:29 - Connection State - Disconnected
2025-06-06 14:17:29 - Cleared all pending method requests due to disconnect
2025-06-06 14:17:29 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:29 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:17:29 - Connect using port 8883 (TCP)
2025-06-06 14:17:32 - disconnected with result code: 7
2025-06-06 14:17:32 - connected with result code: 0
2025-06-06 14:17:32 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:17:32 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:17:32 - _on_mqtt_connected called
2025-06-06 14:17:32 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:17:32 - Connection State - Disconnected
2025-06-06 14:17:32 - Cleared all pending method requests due to disconnect
2025-06-06 14:17:32 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:32 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:17:32 - Connection State - Connected
2025-06-06 14:17:32 - Connect using port 8883 (TCP)
2025-06-06 14:17:32 - connected with result code: 0
2025-06-06 14:17:32 - _on_mqtt_connected called
2025-06-06 14:17:32 - Connection State - Connected
2025-06-06 14:17:33 - disconnected with result code: 7
2025-06-06 14:17:33 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:17:33 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:17:33 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:17:33 - Connection State - Disconnected
2025-06-06 14:17:33 - Cleared all pending method requests due to disconnect
2025-06-06 14:17:33 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:33 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:17:33 - Connect using port 8883 (TCP)
2025-06-06 14:17:33 - message received on $iothub/methods/POST/Agitator/?$rid=1
2025-06-06 14:17:33 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:33 - ['azure.iot.device.iothub.sync_handler_manager.HandlerManagerException: HANDLER (_on_method_request_received): Error during invocation\n']
2025-06-06 14:17:36 - connected with result code: 0
2025-06-06 14:17:36 - _on_mqtt_connected called
2025-06-06 14:17:36 - Connection State - Connected
2025-06-06 14:17:36 - disconnected with result code: 7
2025-06-06 14:17:36 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:17:36 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:17:36 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:17:36 - Connection State - Disconnected
2025-06-06 14:17:36 - Cleared all pending method requests due to disconnect
2025-06-06 14:17:36 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:36 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:17:36 - Connect using port 8883 (TCP)
2025-06-06 14:17:38 - connected with result code: 0
2025-06-06 14:17:38 - _on_mqtt_connected called
2025-06-06 14:17:38 - Connection State - Connected
2025-06-06 14:17:38 - disconnected with result code: 7
2025-06-06 14:17:38 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:17:38 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:17:38 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:17:38 - Connection State - Disconnected
2025-06-06 14:17:38 - Cleared all pending method requests due to disconnect
2025-06-06 14:17:38 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:38 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:17:38 - Connect using port 8883 (TCP)
2025-06-06 14:17:39 - disconnected with result code: 7
2025-06-06 14:17:39 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:17:39 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:17:39 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:17:39 - Connection State - Disconnected
2025-06-06 14:17:39 - Cleared all pending method requests due to disconnect
2025-06-06 14:17:39 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:39 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:17:39 - Connect using port 8883 (TCP)
2025-06-06 14:17:39 - connected with result code: 0
2025-06-06 14:17:39 - _on_mqtt_connected called
2025-06-06 14:17:39 - Connection State - Connected
2025-06-06 14:17:40 - connected with result code: 0
2025-06-06 14:17:40 - _on_mqtt_connected called
2025-06-06 14:17:40 - Connection State - Connected
2025-06-06 14:17:40 - disconnected with result code: 7
2025-06-06 14:17:40 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:17:40 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:17:40 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:17:40 - Connection State - Disconnected
2025-06-06 14:17:40 - Cleared all pending method requests due to disconnect
2025-06-06 14:17:40 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:40 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:17:40 - Connect using port 8883 (TCP)
2025-06-06 14:17:41 - disconnected with result code: 7
2025-06-06 14:17:41 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:17:41 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:17:41 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:17:41 - Connection State - Disconnected
2025-06-06 14:17:41 - Cleared all pending method requests due to disconnect
2025-06-06 14:17:41 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:41 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:17:41 - connected with result code: 0
2025-06-06 14:17:41 - _on_mqtt_connected called
2025-06-06 14:17:41 - Connection State - Connected
2025-06-06 14:17:41 - Connect using port 8883 (TCP)
2025-06-06 14:17:42 - connected with result code: 0
2025-06-06 14:17:42 - _on_mqtt_connected called
2025-06-06 14:17:42 - Connection State - Connected
2025-06-06 14:17:42 - disconnected with result code: 7
2025-06-06 14:17:42 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:17:42 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:17:42 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:17:42 - Connection State - Disconnected
2025-06-06 14:17:42 - Cleared all pending method requests due to disconnect
2025-06-06 14:17:42 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:42 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:17:42 - Connect using port 8883 (TCP)
2025-06-06 14:17:43 - connected with result code: 0
2025-06-06 14:17:43 - _on_mqtt_connected called
2025-06-06 14:17:43 - Connection State - Connected
2025-06-06 14:17:43 - disconnected with result code: 7
2025-06-06 14:17:43 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:17:43 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:17:43 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:17:43 - Connection State - Disconnected
2025-06-06 14:17:43 - Cleared all pending method requests due to disconnect
2025-06-06 14:17:43 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:43 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:17:43 - Connect using port 8883 (TCP)
2025-06-06 14:17:44 - connected with result code: 0
2025-06-06 14:17:44 - _on_mqtt_connected called
2025-06-06 14:17:44 - disconnected with result code: 7
2025-06-06 14:17:44 - Connection State - Connected
2025-06-06 14:17:44 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:17:44 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:17:44 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:17:44 - Connection State - Disconnected
2025-06-06 14:17:44 - Cleared all pending method requests due to disconnect
2025-06-06 14:17:44 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:44 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:17:44 - Connect using port 8883 (TCP)
2025-06-06 14:17:45 - disconnected with result code: 7
2025-06-06 14:17:45 - connected with result code: 0
2025-06-06 14:17:45 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:17:45 - _on_mqtt_connected called
2025-06-06 14:17:45 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:17:45 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:17:45 - Connection State - Disconnected
2025-06-06 14:17:45 - Cleared all pending method requests due to disconnect
2025-06-06 14:17:45 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:45 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:17:45 - Connection State - Connected
2025-06-06 14:17:45 - Connect using port 8883 (TCP)
2025-06-06 14:17:46 - connected with result code: 0
2025-06-06 14:17:46 - _on_mqtt_connected called
2025-06-06 14:17:46 - Connection State - Connected
2025-06-06 14:17:46 - disconnected with result code: 7
2025-06-06 14:17:46 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:17:46 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:17:46 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:17:46 - Connection State - Disconnected
2025-06-06 14:17:46 - Cleared all pending method requests due to disconnect
2025-06-06 14:17:46 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:46 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:17:46 - Connect using port 8883 (TCP)
2025-06-06 14:17:47 - disconnected with result code: 7
2025-06-06 14:17:47 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:17:47 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:17:47 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:17:47 - Connection State - Disconnected
2025-06-06 14:17:47 - Cleared all pending method requests due to disconnect
2025-06-06 14:17:47 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:47 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:17:47 - Connect using port 8883 (TCP)
2025-06-06 14:17:47 - connected with result code: 0
2025-06-06 14:17:47 - _on_mqtt_connected called
2025-06-06 14:17:47 - Connection State - Connected
2025-06-06 14:17:49 - connected with result code: 0
2025-06-06 14:17:49 - _on_mqtt_connected called
2025-06-06 14:17:49 - Connection State - Connected
2025-06-06 14:17:49 - disconnected with result code: 7
2025-06-06 14:17:49 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:17:49 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:17:49 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:17:49 - Connection State - Disconnected
2025-06-06 14:17:49 - Cleared all pending method requests due to disconnect
2025-06-06 14:17:49 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:49 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:17:49 - Connect using port 8883 (TCP)
2025-06-06 14:17:54 - connected with result code: 0
2025-06-06 14:17:54 - _on_mqtt_connected called
2025-06-06 14:17:54 - Connection State - Connected
2025-06-06 14:17:54 - disconnected with result code: 7
2025-06-06 14:17:54 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:17:54 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:17:54 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:17:54 - Connection State - Disconnected
2025-06-06 14:17:54 - Cleared all pending method requests due to disconnect
2025-06-06 14:17:54 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:17:54 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:17:54 - Connect using port 8883 (TCP)
2025-06-06 14:18:00 - connected with result code: 0
2025-06-06 14:18:00 - _on_mqtt_connected called
2025-06-06 14:18:00 - disconnected with result code: 7
2025-06-06 14:18:00 - Connection State - Connected
2025-06-06 14:18:00 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:00 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:00 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:00 - Connection State - Disconnected
2025-06-06 14:18:00 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:00 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:00 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:00 - Connect using port 8883 (TCP)
2025-06-06 14:18:06 - Creating client for connecting using MQTT over TCP
2025-06-06 14:18:06 - Connecting to Hub...
2025-06-06 14:18:06 - Connect using port 8883 (TCP)
2025-06-06 14:18:09 - disconnected with result code: 7
2025-06-06 14:18:09 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:09 - connected with result code: 0
2025-06-06 14:18:09 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:09 - _on_mqtt_connected called
2025-06-06 14:18:09 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:09 - Connection State - Disconnected
2025-06-06 14:18:09 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:09 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:09 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:09 - Connection State - Connected
2025-06-06 14:18:09 - Successfully connected to Hub
2025-06-06 14:18:09 - Enabling feature:methods...
2025-06-06 14:18:09 - Connect using port 8883 (TCP)
2025-06-06 14:18:09 - subscribing to $iothub/methods/POST/# with qos 1
2025-06-06 14:18:10 - suback received for 1
2025-06-06 14:18:10 - Successfully enabled feature:methods
2025-06-06 14:18:10 - Khởi động hệ thống
2025-06-06 14:18:10 - connected with result code: 0
2025-06-06 14:18:10 - _on_mqtt_connected called
2025-06-06 14:18:10 - Connection State - Connected
2025-06-06 14:18:10 - disconnected with result code: 7
2025-06-06 14:18:10 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:10 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:10 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:10 - Connection State - Disconnected
2025-06-06 14:18:10 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:10 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:10 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:10 - Connect using port 8883 (TCP)
2025-06-06 14:18:13 - connected with result code: 0
2025-06-06 14:18:13 - _on_mqtt_connected called
2025-06-06 14:18:13 - Connection State - Connected
2025-06-06 14:18:13 - disconnected with result code: 7
2025-06-06 14:18:13 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:13 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:13 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:13 - Connection State - Disconnected
2025-06-06 14:18:13 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:13 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:13 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:13 - Connect using port 8883 (TCP)
2025-06-06 14:18:14 - connected with result code: 0
2025-06-06 14:18:14 - _on_mqtt_connected called
2025-06-06 14:18:14 - disconnected with result code: 7
2025-06-06 14:18:14 - Connection State - Connected
2025-06-06 14:18:14 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:14 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:14 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:14 - Connection State - Disconnected
2025-06-06 14:18:14 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:14 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:14 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:14 - Connect using port 8883 (TCP)
2025-06-06 14:18:17 - disconnected with result code: 7
2025-06-06 14:18:17 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:17 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:17 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:17 - Connection State - Disconnected
2025-06-06 14:18:17 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:17 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:17 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:17 - Connect using port 8883 (TCP)
2025-06-06 14:18:17 - connected with result code: 0
2025-06-06 14:18:17 - _on_mqtt_connected called
2025-06-06 14:18:17 - Connection State - Connected
2025-06-06 14:18:19 - connected with result code: 0
2025-06-06 14:18:19 - _on_mqtt_connected called
2025-06-06 14:18:19 - Connection State - Connected
2025-06-06 14:18:19 - disconnected with result code: 7
2025-06-06 14:18:19 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:19 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:19 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:19 - Connection State - Disconnected
2025-06-06 14:18:19 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:19 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:19 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:19 - Connect using port 8883 (TCP)
2025-06-06 14:18:21 - disconnected with result code: 7
2025-06-06 14:18:21 - connected with result code: 0
2025-06-06 14:18:21 - _on_mqtt_connected called
2025-06-06 14:18:21 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:21 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:21 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:21 - Connection State - Connected
2025-06-06 14:18:21 - Connection State - Disconnected
2025-06-06 14:18:21 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:21 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:21 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:21 - Connect using port 8883 (TCP)
2025-06-06 14:18:21 - connected with result code: 0
2025-06-06 14:18:21 - _on_mqtt_connected called
2025-06-06 14:18:21 - Connection State - Connected
2025-06-06 14:18:22 - disconnected with result code: 7
2025-06-06 14:18:22 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:22 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:22 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:22 - Connection State - Disconnected
2025-06-06 14:18:22 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:22 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:22 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:22 - Connect using port 8883 (TCP)
2025-06-06 14:18:23 - message received on $iothub/methods/POST/Agitator/?$rid=1
2025-06-06 14:18:23 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:23 - ['azure.iot.device.iothub.sync_handler_manager.HandlerManagerException: HANDLER (_on_method_request_received): Error during invocation\n']
2025-06-06 14:18:28 - connected with result code: 0
2025-06-06 14:18:28 - _on_mqtt_connected called
2025-06-06 14:18:28 - Connection State - Connected
2025-06-06 14:18:29 - disconnected with result code: 7
2025-06-06 14:18:29 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:29 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:29 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:29 - Connection State - Disconnected
2025-06-06 14:18:29 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:29 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:29 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:29 - Connect using port 8883 (TCP)
2025-06-06 14:18:29 - connected with result code: 0
2025-06-06 14:18:29 - _on_mqtt_connected called
2025-06-06 14:18:29 - Connection State - Connected
2025-06-06 14:18:29 - disconnected with result code: 7
2025-06-06 14:18:29 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:29 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:29 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:29 - Connection State - Disconnected
2025-06-06 14:18:29 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:29 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:29 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:30 - Connect using port 8883 (TCP)
2025-06-06 14:18:35 - disconnected with result code: 7
2025-06-06 14:18:35 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:35 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:35 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:35 - Connection State - Disconnected
2025-06-06 14:18:35 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:35 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:35 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:35 - Connect using port 8883 (TCP)
2025-06-06 14:18:35 - connected with result code: 0
2025-06-06 14:18:35 - _on_mqtt_connected called
2025-06-06 14:18:35 - Connection State - Connected
2025-06-06 14:18:37 - connected with result code: 0
2025-06-06 14:18:37 - _on_mqtt_connected called
2025-06-06 14:18:37 - Connection State - Connected
2025-06-06 14:18:37 - disconnected with result code: 7
2025-06-06 14:18:37 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:37 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:37 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:37 - Connection State - Disconnected
2025-06-06 14:18:37 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:37 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:37 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:37 - Connect using port 8883 (TCP)
2025-06-06 14:18:38 - connected with result code: 0
2025-06-06 14:18:38 - _on_mqtt_connected called
2025-06-06 14:18:38 - Connection State - Connected
2025-06-06 14:18:38 - disconnected with result code: 7
2025-06-06 14:18:38 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:38 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:38 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:38 - Connection State - Disconnected
2025-06-06 14:18:38 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:38 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:38 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:38 - Connect using port 8883 (TCP)
2025-06-06 14:18:40 - connected with result code: 0
2025-06-06 14:18:40 - _on_mqtt_connected called
2025-06-06 14:18:40 - Connection State - Connected
2025-06-06 14:18:40 - disconnected with result code: 7
2025-06-06 14:18:40 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:40 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:40 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:40 - Connection State - Disconnected
2025-06-06 14:18:40 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:40 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:40 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:40 - Connect using port 8883 (TCP)
2025-06-06 14:18:42 - disconnected with result code: 7
2025-06-06 14:18:42 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:42 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:42 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:42 - Connection State - Disconnected
2025-06-06 14:18:42 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:42 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:42 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:42 - Connect using port 8883 (TCP)
2025-06-06 14:18:42 - connected with result code: 0
2025-06-06 14:18:42 - _on_mqtt_connected called
2025-06-06 14:18:42 - Connection State - Connected
2025-06-06 14:18:43 - connected with result code: 0
2025-06-06 14:18:43 - _on_mqtt_connected called
2025-06-06 14:18:43 - Connection State - Connected
2025-06-06 14:18:44 - disconnected with result code: 7
2025-06-06 14:18:44 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:44 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:44 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:44 - Connection State - Disconnected
2025-06-06 14:18:44 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:44 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:44 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:44 - Connect using port 8883 (TCP)
2025-06-06 14:18:45 - disconnected with result code: 7
2025-06-06 14:18:45 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:45 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:45 - connected with result code: 0
2025-06-06 14:18:45 - _on_mqtt_connected called
2025-06-06 14:18:45 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:45 - Connection State - Disconnected
2025-06-06 14:18:45 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:45 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:45 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:45 - Connect using port 8883 (TCP)
2025-06-06 14:18:45 - Connection State - Connected
2025-06-06 14:18:46 - connected with result code: 0
2025-06-06 14:18:46 - _on_mqtt_connected called
2025-06-06 14:18:46 - Connection State - Connected
2025-06-06 14:18:46 - disconnected with result code: 7
2025-06-06 14:18:46 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:46 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:46 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:46 - Connection State - Disconnected
2025-06-06 14:18:46 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:46 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:46 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:46 - Connect using port 8883 (TCP)
2025-06-06 14:18:48 - connected with result code: 0
2025-06-06 14:18:48 - _on_mqtt_connected called
2025-06-06 14:18:48 - Connection State - Connected
2025-06-06 14:18:48 - disconnected with result code: 7
2025-06-06 14:18:48 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:48 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:48 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:48 - Connection State - Disconnected
2025-06-06 14:18:48 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:48 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:48 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:48 - Connect using port 8883 (TCP)
2025-06-06 14:18:48 - connected with result code: 0
2025-06-06 14:18:48 - _on_mqtt_connected called
2025-06-06 14:18:48 - Connection State - Connected
2025-06-06 14:18:48 - disconnected with result code: 7
2025-06-06 14:18:48 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:48 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:48 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:48 - Connection State - Disconnected
2025-06-06 14:18:48 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:48 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:48 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:48 - Connect using port 8883 (TCP)
2025-06-06 14:18:51 - disconnected with result code: 7
2025-06-06 14:18:51 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:51 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:51 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:51 - Connection State - Disconnected
2025-06-06 14:18:51 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:51 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:51 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:51 - connected with result code: 0
2025-06-06 14:18:51 - _on_mqtt_connected called
2025-06-06 14:18:51 - Connection State - Connected
2025-06-06 14:18:51 - Connect using port 8883 (TCP)
2025-06-06 14:18:53 - connected with result code: 0
2025-06-06 14:18:53 - _on_mqtt_connected called
2025-06-06 14:18:53 - Connection State - Connected
2025-06-06 14:18:57 - Creating client for connecting using MQTT over TCP
2025-06-06 14:18:57 - Connecting to Hub...
2025-06-06 14:18:57 - Connect using port 8883 (TCP)
2025-06-06 14:18:58 - disconnected with result code: 7
2025-06-06 14:18:59 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:18:59 - connected with result code: 0
2025-06-06 14:18:59 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:18:59 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:18:59 - _on_mqtt_connected called
2025-06-06 14:18:59 - Connection State - Disconnected
2025-06-06 14:18:59 - Cleared all pending method requests due to disconnect
2025-06-06 14:18:59 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:18:59 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:18:59 - Connection State - Connected
2025-06-06 14:18:59 - Successfully connected to Hub
2025-06-06 14:18:59 - Enabling feature:methods...
2025-06-06 14:18:59 - Connect using port 8883 (TCP)
2025-06-06 14:18:59 - subscribing to $iothub/methods/POST/# with qos 1
2025-06-06 14:18:59 - suback received for 1
2025-06-06 14:18:59 - Successfully enabled feature:methods
2025-06-06 14:18:59 - Khởi động hệ thống
2025-06-06 14:19:00 - connected with result code: 0
2025-06-06 14:19:00 - _on_mqtt_connected called
2025-06-06 14:19:00 - Connection State - Connected
2025-06-06 14:19:00 - disconnected with result code: 7
2025-06-06 14:19:00 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:19:00 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:19:00 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:19:00 - Connection State - Disconnected
2025-06-06 14:19:00 - Cleared all pending method requests due to disconnect
2025-06-06 14:19:00 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:19:00 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:19:00 - Connect using port 8883 (TCP)
2025-06-06 14:19:00 - connected with result code: 0
2025-06-06 14:19:00 - _on_mqtt_connected called
2025-06-06 14:19:00 - Connection State - Connected
2025-06-06 14:19:00 - disconnected with result code: 7
2025-06-06 14:19:00 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:19:00 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:19:00 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:19:00 - Connection State - Disconnected
2025-06-06 14:19:00 - Cleared all pending method requests due to disconnect
2025-06-06 14:19:00 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:19:00 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:19:00 - Connect using port 8883 (TCP)
2025-06-06 14:19:00 - connected with result code: 0
2025-06-06 14:19:00 - _on_mqtt_connected called
2025-06-06 14:19:00 - Connection State - Connected
2025-06-06 14:19:01 - disconnected with result code: 7
2025-06-06 14:19:01 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:19:01 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:19:01 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:19:01 - Connection State - Disconnected
2025-06-06 14:19:01 - Cleared all pending method requests due to disconnect
2025-06-06 14:19:01 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:19:01 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:19:01 - Connect using port 8883 (TCP)
2025-06-06 14:19:01 - connected with result code: 0
2025-06-06 14:19:01 - _on_mqtt_connected called
2025-06-06 14:19:01 - Connection State - Connected
2025-06-06 14:19:01 - disconnected with result code: 7
2025-06-06 14:19:01 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:19:01 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:19:01 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:19:01 - Connection State - Disconnected
2025-06-06 14:19:01 - Cleared all pending method requests due to disconnect
2025-06-06 14:19:01 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:19:01 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:19:01 - Connect using port 8883 (TCP)
2025-06-06 14:19:04 - connected with result code: 0
2025-06-06 14:19:04 - _on_mqtt_connected called
2025-06-06 14:19:04 - Connection State - Connected
2025-06-06 14:19:04 - disconnected with result code: 7
2025-06-06 14:19:04 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:19:04 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:19:04 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:19:04 - Connection State - Disconnected
2025-06-06 14:19:04 - Cleared all pending method requests due to disconnect
2025-06-06 14:19:04 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:19:04 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:19:04 - Connect using port 8883 (TCP)
2025-06-06 14:19:10 - connected with result code: 0
2025-06-06 14:19:10 - _on_mqtt_connected called
2025-06-06 14:19:10 - Connection State - Connected
2025-06-06 14:19:10 - disconnected with result code: 7
2025-06-06 14:19:10 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:19:10 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:19:10 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:19:10 - Connection State - Disconnected
2025-06-06 14:19:10 - Cleared all pending method requests due to disconnect
2025-06-06 14:19:10 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:19:10 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:19:10 - Connect using port 8883 (TCP)
2025-06-06 14:19:10 - message received on $iothub/methods/POST/Agitator/?$rid=1
2025-06-06 14:19:11 - connected with result code: 0
2025-06-06 14:19:11 - _on_mqtt_connected called
2025-06-06 14:19:11 - Connection State - Connected
2025-06-06 14:19:19 - disconnected with result code: 7
2025-06-06 14:19:19 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:19:19 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:19:19 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:19:19 - Connection State - Disconnected
2025-06-06 14:19:19 - Cleared all pending method requests due to disconnect
2025-06-06 14:19:19 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:19:19 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:19:19 - Connect using port 8883 (TCP)
2025-06-06 14:19:30 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:19:30 - ['azure.iot.device.iothub.sync_handler_manager.HandlerManagerException: HANDLER (_on_method_request_received): Error during invocation\n']
2025-06-06 14:19:32 - disconnected with result code: 7
2025-06-06 14:19:32 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:19:32 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:19:32 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:19:32 - Connection State - Disconnected
2025-06-06 14:19:32 - Cleared all pending method requests due to disconnect
2025-06-06 14:19:32 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:19:32 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:19:32 - Connect using port 8883 (TCP)
2025-06-06 14:19:32 - connected with result code: 0
2025-06-06 14:19:32 - _on_mqtt_connected called
2025-06-06 14:19:32 - Connection State - Connected
2025-06-06 14:19:38 - connected with result code: 0
2025-06-06 14:19:38 - _on_mqtt_connected called
2025-06-06 14:19:38 - Connection State - Connected
2025-06-06 14:19:38 - disconnected with result code: 7
2025-06-06 14:19:38 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:19:38 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:19:38 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:19:38 - Connection State - Disconnected
2025-06-06 14:19:38 - Cleared all pending method requests due to disconnect
2025-06-06 14:19:38 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:19:38 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:19:38 - Connect using port 8883 (TCP)
2025-06-06 14:19:39 - message received on $iothub/methods/POST/Agitator/?$rid=1
2025-06-06 14:19:39 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:19:39 - ['azure.iot.device.iothub.sync_handler_manager.HandlerManagerException: HANDLER (_on_method_request_received): Error during invocation\n']
2025-06-06 14:19:51 - disconnected with result code: 7
2025-06-06 14:19:51 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:19:51 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:19:51 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:19:51 - Connection State - Disconnected
2025-06-06 14:19:51 - Cleared all pending method requests due to disconnect
2025-06-06 14:19:51 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:19:51 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:19:51 - Connect using port 8883 (TCP)
2025-06-06 14:20:11 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:20:11 - transport.connect raised error
2025-06-06 14:20:11 - Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 396, in connect
    rc = self._mqtt_client.connect(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 914, in connect
    return self.reconnect()
           ^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 1044, in reconnect
    sock = self._create_socket_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 3685, in _create_socket_connection
    return socket.create_connection(addr, timeout=self._connect_timeout, source_address=source)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 827, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 962, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
socket.gaierror: [Errno -3] Temporary failure in name resolution

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py", line 206, in _run_op
    self.transport.connect(password=password)
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 419, in connect
    raise exceptions.ConnectionFailedError() from e
azure.iot.device.common.transport_exceptions.ConnectionFailedError

2025-06-06 14:20:11 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:20:11 - ['azure.iot.device.common.transport_exceptions.ConnectionFailedError\n']
2025-06-06 14:20:21 - Connect using port 8883 (TCP)
2025-06-06 14:20:39 - MQTTTransportStage(ConnectOperation): Connection watchdog expired.  Cancelling op
2025-06-06 14:20:39 - disconnecting MQTT client
2025-06-06 14:20:39 - disconnected with result code: 0
2025-06-06 14:20:39 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:20:39 - ['azure.iot.device.common.pipeline.pipeline_exceptions.OperationTimeout: Transport timeout on connection operation\n']
2025-06-06 14:20:39 - MQTTTransportStage: _on_mqtt_disconnect called
2025-06-06 14:20:39 - ConnectionStateStage: DisconnectEvent received while in unexpected state - ConnectionState.DISCONNECTED
2025-06-06 14:20:39 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:20:39 - Connection State - Disconnected
2025-06-06 14:20:39 - Cleared all pending method requests due to disconnect
2025-06-06 14:20:39 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:20:39 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:20:41 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:20:41 - transport.connect raised error
2025-06-06 14:20:41 - Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 396, in connect
    rc = self._mqtt_client.connect(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 914, in connect
    return self.reconnect()
           ^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 1044, in reconnect
    sock = self._create_socket_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 3685, in _create_socket_connection
    return socket.create_connection(addr, timeout=self._connect_timeout, source_address=source)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 827, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 962, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
socket.gaierror: [Errno -3] Temporary failure in name resolution

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py", line 206, in _run_op
    self.transport.connect(password=password)
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 419, in connect
    raise exceptions.ConnectionFailedError() from e
azure.iot.device.common.transport_exceptions.ConnectionFailedError

2025-06-06 14:20:41 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:20:41 - ['azure.iot.device.common.transport_exceptions.ConnectionFailedError\n']
2025-06-06 14:20:49 - Connect using port 8883 (TCP)
2025-06-06 14:20:51 - Connect using port 8883 (TCP)
2025-06-06 14:21:09 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:21:09 - transport.connect raised error
2025-06-06 14:21:09 - Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 396, in connect
    rc = self._mqtt_client.connect(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 914, in connect
    return self.reconnect()
           ^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 1044, in reconnect
    sock = self._create_socket_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 3685, in _create_socket_connection
    return socket.create_connection(addr, timeout=self._connect_timeout, source_address=source)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 827, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 962, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
socket.gaierror: [Errno -3] Temporary failure in name resolution

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py", line 206, in _run_op
    self.transport.connect(password=password)
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 419, in connect
    raise exceptions.ConnectionFailedError() from e
azure.iot.device.common.transport_exceptions.ConnectionFailedError

2025-06-06 14:21:09 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:21:09 - ['azure.iot.device.common.transport_exceptions.ConnectionFailedError\n']
2025-06-06 14:21:11 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:21:11 - transport.connect raised error
2025-06-06 14:21:11 - Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 396, in connect
    rc = self._mqtt_client.connect(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 914, in connect
    return self.reconnect()
           ^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 1044, in reconnect
    sock = self._create_socket_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 3685, in _create_socket_connection
    return socket.create_connection(addr, timeout=self._connect_timeout, source_address=source)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 827, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 962, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
socket.gaierror: [Errno -3] Temporary failure in name resolution

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py", line 206, in _run_op
    self.transport.connect(password=password)
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 419, in connect
    raise exceptions.ConnectionFailedError() from e
azure.iot.device.common.transport_exceptions.ConnectionFailedError

2025-06-06 14:21:11 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:21:11 - ['azure.iot.device.common.transport_exceptions.ConnectionFailedError\n']
2025-06-06 14:21:19 - Connect using port 8883 (TCP)
2025-06-06 14:21:21 - Connect using port 8883 (TCP)
2025-06-06 14:21:39 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:21:39 - transport.connect raised error
2025-06-06 14:21:39 - Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 396, in connect
    rc = self._mqtt_client.connect(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 914, in connect
    return self.reconnect()
           ^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 1044, in reconnect
    sock = self._create_socket_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 3685, in _create_socket_connection
    return socket.create_connection(addr, timeout=self._connect_timeout, source_address=source)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 827, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 962, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
socket.gaierror: [Errno -3] Temporary failure in name resolution

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py", line 206, in _run_op
    self.transport.connect(password=password)
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 419, in connect
    raise exceptions.ConnectionFailedError() from e
azure.iot.device.common.transport_exceptions.ConnectionFailedError

2025-06-06 14:21:39 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:21:39 - ['azure.iot.device.common.transport_exceptions.ConnectionFailedError\n']
2025-06-06 14:21:41 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:21:41 - transport.connect raised error
2025-06-06 14:21:41 - Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 396, in connect
    rc = self._mqtt_client.connect(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 914, in connect
    return self.reconnect()
           ^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 1044, in reconnect
    sock = self._create_socket_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 3685, in _create_socket_connection
    return socket.create_connection(addr, timeout=self._connect_timeout, source_address=source)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 827, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 962, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
socket.gaierror: [Errno -3] Temporary failure in name resolution

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py", line 206, in _run_op
    self.transport.connect(password=password)
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 419, in connect
    raise exceptions.ConnectionFailedError() from e
azure.iot.device.common.transport_exceptions.ConnectionFailedError

2025-06-06 14:21:41 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:21:41 - ['azure.iot.device.common.transport_exceptions.ConnectionFailedError\n']
2025-06-06 14:21:49 - Connect using port 8883 (TCP)
2025-06-06 14:21:51 - Connect using port 8883 (TCP)
2025-06-06 14:22:09 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:22:09 - transport.connect raised error
2025-06-06 14:22:09 - Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 396, in connect
    rc = self._mqtt_client.connect(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 914, in connect
    return self.reconnect()
           ^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 1044, in reconnect
    sock = self._create_socket_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 3685, in _create_socket_connection
    return socket.create_connection(addr, timeout=self._connect_timeout, source_address=source)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 827, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 962, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
socket.gaierror: [Errno -3] Temporary failure in name resolution

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py", line 206, in _run_op
    self.transport.connect(password=password)
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 419, in connect
    raise exceptions.ConnectionFailedError() from e
azure.iot.device.common.transport_exceptions.ConnectionFailedError

2025-06-06 14:22:09 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:22:09 - ['azure.iot.device.common.transport_exceptions.ConnectionFailedError\n']
2025-06-06 14:22:11 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:22:11 - transport.connect raised error
2025-06-06 14:22:11 - Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 396, in connect
    rc = self._mqtt_client.connect(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 914, in connect
    return self.reconnect()
           ^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 1044, in reconnect
    sock = self._create_socket_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 3685, in _create_socket_connection
    return socket.create_connection(addr, timeout=self._connect_timeout, source_address=source)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 827, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 962, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
socket.gaierror: [Errno -3] Temporary failure in name resolution

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py", line 206, in _run_op
    self.transport.connect(password=password)
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 419, in connect
    raise exceptions.ConnectionFailedError() from e
azure.iot.device.common.transport_exceptions.ConnectionFailedError

2025-06-06 14:22:11 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:22:11 - ['azure.iot.device.common.transport_exceptions.ConnectionFailedError\n']
2025-06-06 14:22:15 - Creating client for connecting using MQTT over TCP
2025-06-06 14:22:15 - Connecting to Hub...
2025-06-06 14:22:15 - Connect using port 8883 (TCP)
2025-06-06 14:22:21 - Connect using port 8883 (TCP)
2025-06-06 14:22:39 - Creating client for connecting using MQTT over TCP
2025-06-06 14:22:39 - Connecting to Hub...
2025-06-06 14:22:39 - Connect using port 8883 (TCP)
2025-06-06 14:22:41 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:22:41 - transport.connect raised error
2025-06-06 14:22:41 - Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 396, in connect
    rc = self._mqtt_client.connect(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 914, in connect
    return self.reconnect()
           ^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 1044, in reconnect
    sock = self._create_socket_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 3685, in _create_socket_connection
    return socket.create_connection(addr, timeout=self._connect_timeout, source_address=source)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 827, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 962, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
socket.gaierror: [Errno -3] Temporary failure in name resolution

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py", line 206, in _run_op
    self.transport.connect(password=password)
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 419, in connect
    raise exceptions.ConnectionFailedError() from e
azure.iot.device.common.transport_exceptions.ConnectionFailedError

2025-06-06 14:22:41 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:22:41 - ['azure.iot.device.common.transport_exceptions.ConnectionFailedError\n']
2025-06-06 14:22:51 - Connect using port 8883 (TCP)
2025-06-06 14:22:59 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:22:59 - transport.connect raised error
2025-06-06 14:22:59 - Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 396, in connect
    rc = self._mqtt_client.connect(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 914, in connect
    return self.reconnect()
           ^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 1044, in reconnect
    sock = self._create_socket_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 3685, in _create_socket_connection
    return socket.create_connection(addr, timeout=self._connect_timeout, source_address=source)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 827, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 962, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
socket.gaierror: [Errno -3] Temporary failure in name resolution

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py", line 206, in _run_op
    self.transport.connect(password=password)
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 419, in connect
    raise exceptions.ConnectionFailedError() from e
azure.iot.device.common.transport_exceptions.ConnectionFailedError

2025-06-06 14:22:59 - Callback completed with error 
2025-06-06 14:22:59 - NoneType: None

2025-06-06 14:23:11 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:23:11 - transport.connect raised error
2025-06-06 14:23:11 - Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 396, in connect
    rc = self._mqtt_client.connect(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 914, in connect
    return self.reconnect()
           ^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 1044, in reconnect
    sock = self._create_socket_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 3685, in _create_socket_connection
    return socket.create_connection(addr, timeout=self._connect_timeout, source_address=source)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 827, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 962, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
socket.gaierror: [Errno -3] Temporary failure in name resolution

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py", line 206, in _run_op
    self.transport.connect(password=password)
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 419, in connect
    raise exceptions.ConnectionFailedError() from e
azure.iot.device.common.transport_exceptions.ConnectionFailedError

2025-06-06 14:23:11 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:23:11 - ['azure.iot.device.common.transport_exceptions.ConnectionFailedError\n']
2025-06-06 14:23:21 - Connect using port 8883 (TCP)
2025-06-06 14:23:41 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:23:41 - transport.connect raised error
2025-06-06 14:23:41 - Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 396, in connect
    rc = self._mqtt_client.connect(
         ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 914, in connect
    return self.reconnect()
           ^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 1044, in reconnect
    sock = self._create_socket_connection()
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 3685, in _create_socket_connection
    return socket.create_connection(addr, timeout=self._connect_timeout, source_address=source)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 827, in create_connection
    for res in getaddrinfo(host, port, 0, SOCK_STREAM):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/socket.py", line 962, in getaddrinfo
    for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
socket.gaierror: [Errno -3] Temporary failure in name resolution

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py", line 206, in _run_op
    self.transport.connect(password=password)
  File "/home/admin/Desktop/works/venv/lib/python3.11/site-packages/azure/iot/device/common/mqtt_transport.py", line 419, in connect
    raise exceptions.ConnectionFailedError() from e
azure.iot.device.common.transport_exceptions.ConnectionFailedError

2025-06-06 14:23:41 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:23:41 - ['azure.iot.device.common.transport_exceptions.ConnectionFailedError\n']
2025-06-06 14:23:51 - Connect using port 8883 (TCP)
2025-06-06 14:23:52 - connected with result code: 0
2025-06-06 14:23:52 - _on_mqtt_connected called
2025-06-06 14:23:52 - Connection State - Connected
2025-06-06 14:25:35 - Creating client for connecting using MQTT over TCP
2025-06-06 14:25:35 - Connecting to Hub...
2025-06-06 14:25:35 - Connect using port 8883 (TCP)
2025-06-06 14:25:36 - disconnected with result code: 7
2025-06-06 14:25:36 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:36 - connected with result code: 0
2025-06-06 14:25:36 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:36 - _on_mqtt_connected called
2025-06-06 14:25:36 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:36 - Connection State - Disconnected
2025-06-06 14:25:36 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:36 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:36 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:36 - Connection State - Connected
2025-06-06 14:25:36 - Successfully connected to Hub
2025-06-06 14:25:36 - Enabling feature:methods...
2025-06-06 14:25:36 - Connect using port 8883 (TCP)
2025-06-06 14:25:36 - subscribing to $iothub/methods/POST/# with qos 1
2025-06-06 14:25:36 - suback received for 1
2025-06-06 14:25:36 - Successfully enabled feature:methods
2025-06-06 14:25:36 - Khởi động hệ thống
2025-06-06 14:25:37 - connected with result code: 0
2025-06-06 14:25:37 - _on_mqtt_connected called
2025-06-06 14:25:37 - Connection State - Connected
2025-06-06 14:25:37 - disconnected with result code: 7
2025-06-06 14:25:37 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:37 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:37 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:37 - Connection State - Disconnected
2025-06-06 14:25:37 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:37 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:37 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:37 - Connect using port 8883 (TCP)
2025-06-06 14:25:38 - disconnected with result code: 7
2025-06-06 14:25:38 - connected with result code: 0
2025-06-06 14:25:38 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:38 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:38 - _on_mqtt_connected called
2025-06-06 14:25:38 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:38 - Connection State - Disconnected
2025-06-06 14:25:38 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:38 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:38 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:38 - Connection State - Connected
2025-06-06 14:25:38 - Connect using port 8883 (TCP)
2025-06-06 14:25:38 - connected with result code: 0
2025-06-06 14:25:38 - _on_mqtt_connected called
2025-06-06 14:25:38 - disconnected with result code: 7
2025-06-06 14:25:38 - Connection State - Connected
2025-06-06 14:25:38 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:38 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:38 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:38 - Connection State - Disconnected
2025-06-06 14:25:38 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:38 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:38 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:38 - Connect using port 8883 (TCP)
2025-06-06 14:25:39 - disconnected with result code: 7
2025-06-06 14:25:39 - connected with result code: 0
2025-06-06 14:25:39 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:39 - _on_mqtt_connected called
2025-06-06 14:25:39 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:39 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:39 - Connection State - Connected
2025-06-06 14:25:39 - Connection State - Disconnected
2025-06-06 14:25:39 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:39 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:39 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:39 - Connect using port 8883 (TCP)
2025-06-06 14:25:39 - connected with result code: 0
2025-06-06 14:25:39 - _on_mqtt_connected called
2025-06-06 14:25:39 - disconnected with result code: 7
2025-06-06 14:25:39 - Connection State - Connected
2025-06-06 14:25:39 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:39 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:39 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:39 - Connection State - Disconnected
2025-06-06 14:25:39 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:39 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:39 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:39 - Connect using port 8883 (TCP)
2025-06-06 14:25:40 - connected with result code: 0
2025-06-06 14:25:40 - _on_mqtt_connected called
2025-06-06 14:25:40 - Connection State - Connected
2025-06-06 14:25:40 - disconnected with result code: 7
2025-06-06 14:25:40 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:40 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:40 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:40 - Connection State - Disconnected
2025-06-06 14:25:40 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:40 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:40 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:40 - Connect using port 8883 (TCP)
2025-06-06 14:25:40 - connected with result code: 0
2025-06-06 14:25:40 - _on_mqtt_connected called
2025-06-06 14:25:40 - Connection State - Connected
2025-06-06 14:25:40 - disconnected with result code: 7
2025-06-06 14:25:40 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:40 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:40 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:40 - Connection State - Disconnected
2025-06-06 14:25:40 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:40 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:40 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:40 - Connect using port 8883 (TCP)
2025-06-06 14:25:41 - disconnected with result code: 7
2025-06-06 14:25:41 - connected with result code: 0
2025-06-06 14:25:41 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:41 - _on_mqtt_connected called
2025-06-06 14:25:41 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:41 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:41 - Connection State - Disconnected
2025-06-06 14:25:41 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:41 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:41 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:41 - Connection State - Connected
2025-06-06 14:25:41 - Connect using port 8883 (TCP)
2025-06-06 14:25:41 - connected with result code: 0
2025-06-06 14:25:41 - _on_mqtt_connected called
2025-06-06 14:25:41 - Connection State - Connected
2025-06-06 14:25:41 - disconnected with result code: 7
2025-06-06 14:25:41 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:41 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:41 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:41 - Connection State - Disconnected
2025-06-06 14:25:41 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:41 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:41 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:41 - Connect using port 8883 (TCP)
2025-06-06 14:25:42 - disconnected with result code: 7
2025-06-06 14:25:42 - connected with result code: 0
2025-06-06 14:25:42 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:42 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:42 - _on_mqtt_connected called
2025-06-06 14:25:42 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:42 - Connection State - Disconnected
2025-06-06 14:25:42 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:42 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:42 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:42 - Connection State - Connected
2025-06-06 14:25:42 - Connect using port 8883 (TCP)
2025-06-06 14:25:43 - connected with result code: 0
2025-06-06 14:25:43 - _on_mqtt_connected called
2025-06-06 14:25:43 - disconnected with result code: 7
2025-06-06 14:25:43 - Connection State - Connected
2025-06-06 14:25:43 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:43 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:43 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:43 - Connection State - Disconnected
2025-06-06 14:25:43 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:43 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:43 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:43 - Connect using port 8883 (TCP)
2025-06-06 14:25:43 - connected with result code: 0
2025-06-06 14:25:43 - _on_mqtt_connected called
2025-06-06 14:25:43 - Connection State - Connected
2025-06-06 14:25:43 - disconnected with result code: 7
2025-06-06 14:25:43 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:43 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:43 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:43 - Connection State - Disconnected
2025-06-06 14:25:43 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:43 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:43 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:43 - Connect using port 8883 (TCP)
2025-06-06 14:25:43 - connected with result code: 0
2025-06-06 14:25:43 - _on_mqtt_connected called
2025-06-06 14:25:43 - Connection State - Connected
2025-06-06 14:25:43 - disconnected with result code: 7
2025-06-06 14:25:43 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:43 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:43 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:43 - Connection State - Disconnected
2025-06-06 14:25:43 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:43 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:43 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:43 - Connect using port 8883 (TCP)
2025-06-06 14:25:44 - connected with result code: 0
2025-06-06 14:25:44 - _on_mqtt_connected called
2025-06-06 14:25:44 - Connection State - Connected
2025-06-06 14:25:44 - disconnected with result code: 7
2025-06-06 14:25:44 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:44 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:44 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:44 - Connection State - Disconnected
2025-06-06 14:25:44 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:44 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:44 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:44 - Connect using port 8883 (TCP)
2025-06-06 14:25:44 - connected with result code: 0
2025-06-06 14:25:44 - _on_mqtt_connected called
2025-06-06 14:25:44 - Connection State - Connected
2025-06-06 14:25:44 - disconnected with result code: 7
2025-06-06 14:25:44 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:44 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:44 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:44 - Connection State - Disconnected
2025-06-06 14:25:44 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:44 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:44 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:44 - Connect using port 8883 (TCP)
2025-06-06 14:25:44 - connected with result code: 0
2025-06-06 14:25:44 - disconnected with result code: 7
2025-06-06 14:25:44 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:44 - _on_mqtt_connected called
2025-06-06 14:25:44 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:44 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:44 - Connection State - Disconnected
2025-06-06 14:25:44 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:44 - Connection State - Connected
2025-06-06 14:25:44 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:44 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:44 - Connect using port 8883 (TCP)
2025-06-06 14:25:44 - connected with result code: 0
2025-06-06 14:25:44 - _on_mqtt_connected called
2025-06-06 14:25:44 - disconnected with result code: 7
2025-06-06 14:25:44 - Connection State - Connected
2025-06-06 14:25:44 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:45 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:45 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:45 - Connection State - Disconnected
2025-06-06 14:25:45 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:45 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:45 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:45 - Connect using port 8883 (TCP)
2025-06-06 14:25:45 - disconnected with result code: 7
2025-06-06 14:25:45 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:45 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:45 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:45 - connected with result code: 0
2025-06-06 14:25:45 - Connection State - Disconnected
2025-06-06 14:25:45 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:45 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:45 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:45 - _on_mqtt_connected called
2025-06-06 14:25:45 - Connection State - Connected
2025-06-06 14:25:45 - Connect using port 8883 (TCP)
2025-06-06 14:25:45 - connected with result code: 0
2025-06-06 14:25:45 - _on_mqtt_connected called
2025-06-06 14:25:45 - Connection State - Connected
2025-06-06 14:25:45 - disconnected with result code: 7
2025-06-06 14:25:45 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:45 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:45 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:45 - Connection State - Disconnected
2025-06-06 14:25:45 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:45 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:45 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:45 - Connect using port 8883 (TCP)
2025-06-06 14:25:45 - connected with result code: 0
2025-06-06 14:25:45 - _on_mqtt_connected called
2025-06-06 14:25:45 - Connection State - Connected
2025-06-06 14:25:45 - disconnected with result code: 7
2025-06-06 14:25:45 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:45 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:45 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:45 - Connection State - Disconnected
2025-06-06 14:25:45 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:45 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:45 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:45 - Connect using port 8883 (TCP)
2025-06-06 14:25:46 - disconnected with result code: 7
2025-06-06 14:25:46 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:46 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:46 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:46 - connected with result code: 0
2025-06-06 14:25:46 - Connection State - Disconnected
2025-06-06 14:25:46 - _on_mqtt_connected called
2025-06-06 14:25:46 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:46 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:46 - Connection State - Connected
2025-06-06 14:25:46 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:46 - Connect using port 8883 (TCP)
2025-06-06 14:25:46 - connected with result code: 0
2025-06-06 14:25:46 - _on_mqtt_connected called
2025-06-06 14:25:46 - disconnected with result code: 7
2025-06-06 14:25:46 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:46 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:46 - Connection State - Connected
2025-06-06 14:25:46 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:46 - Connection State - Disconnected
2025-06-06 14:25:46 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:46 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:46 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:46 - Connect using port 8883 (TCP)
2025-06-06 14:25:46 - disconnected with result code: 7
2025-06-06 14:25:46 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:46 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:46 - connected with result code: 0
2025-06-06 14:25:46 - _on_mqtt_connected called
2025-06-06 14:25:46 - Connection State - Connected
2025-06-06 14:25:46 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:46 - Connection State - Disconnected
2025-06-06 14:25:46 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:46 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:46 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:46 - Connect using port 8883 (TCP)
2025-06-06 14:25:47 - connected with result code: 0
2025-06-06 14:25:47 - _on_mqtt_connected called
2025-06-06 14:25:47 - Connection State - Connected
2025-06-06 14:25:47 - disconnected with result code: 7
2025-06-06 14:25:47 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:47 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:47 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:47 - Connection State - Disconnected
2025-06-06 14:25:47 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:47 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:47 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:47 - Connect using port 8883 (TCP)
2025-06-06 14:25:47 - connected with result code: 0
2025-06-06 14:25:47 - _on_mqtt_connected called
2025-06-06 14:25:47 - Connection State - Connected
2025-06-06 14:25:47 - disconnected with result code: 7
2025-06-06 14:25:47 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:47 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:47 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:47 - Connection State - Disconnected
2025-06-06 14:25:47 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:47 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:47 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:47 - Connect using port 8883 (TCP)
2025-06-06 14:25:47 - connected with result code: 0
2025-06-06 14:25:47 - _on_mqtt_connected called
2025-06-06 14:25:47 - disconnected with result code: 7
2025-06-06 14:25:47 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:47 - Connection State - Connected
2025-06-06 14:25:47 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:47 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:47 - Connection State - Disconnected
2025-06-06 14:25:47 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:47 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:47 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:47 - Connect using port 8883 (TCP)
2025-06-06 14:25:48 - disconnected with result code: 7
2025-06-06 14:25:48 - connected with result code: 0
2025-06-06 14:25:48 - _on_mqtt_connected called
2025-06-06 14:25:48 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:48 - Connection State - Connected
2025-06-06 14:25:48 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:48 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:48 - Connection State - Disconnected
2025-06-06 14:25:48 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:48 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:48 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:48 - Connect using port 8883 (TCP)
2025-06-06 14:25:48 - connected with result code: 0
2025-06-06 14:25:48 - _on_mqtt_connected called
2025-06-06 14:25:48 - disconnected with result code: 7
2025-06-06 14:25:48 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:48 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:48 - Connection State - Connected
2025-06-06 14:25:48 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:48 - Connection State - Disconnected
2025-06-06 14:25:48 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:48 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:48 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:48 - Connect using port 8883 (TCP)
2025-06-06 14:25:48 - connected with result code: 0
2025-06-06 14:25:48 - _on_mqtt_connected called
2025-06-06 14:25:48 - Connection State - Connected
2025-06-06 14:25:48 - disconnected with result code: 7
2025-06-06 14:25:48 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:48 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:48 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:48 - Connection State - Disconnected
2025-06-06 14:25:48 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:48 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:48 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:48 - Connect using port 8883 (TCP)
2025-06-06 14:25:48 - connected with result code: 0
2025-06-06 14:25:48 - _on_mqtt_connected called
2025-06-06 14:25:48 - disconnected with result code: 7
2025-06-06 14:25:48 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:48 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:48 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:48 - Connection State - Connected
2025-06-06 14:25:48 - Connection State - Disconnected
2025-06-06 14:25:48 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:48 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:48 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:48 - Connect using port 8883 (TCP)
2025-06-06 14:25:49 - connected with result code: 0
2025-06-06 14:25:49 - _on_mqtt_connected called
2025-06-06 14:25:49 - Connection State - Connected
2025-06-06 14:25:49 - disconnected with result code: 7
2025-06-06 14:25:49 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:49 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:49 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:49 - Connection State - Disconnected
2025-06-06 14:25:49 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:49 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:49 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:49 - Connect using port 8883 (TCP)
2025-06-06 14:25:49 - disconnected with result code: 7
2025-06-06 14:25:49 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:49 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:49 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:49 - Connection State - Disconnected
2025-06-06 14:25:49 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:49 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:49 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:49 - connected with result code: 0
2025-06-06 14:25:49 - _on_mqtt_connected called
2025-06-06 14:25:49 - Connection State - Connected
2025-06-06 14:25:49 - Connect using port 8883 (TCP)
2025-06-06 14:25:49 - connected with result code: 0
2025-06-06 14:25:49 - disconnected with result code: 7
2025-06-06 14:25:49 - _on_mqtt_connected called
2025-06-06 14:25:49 - Connection State - Connected
2025-06-06 14:25:49 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:49 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:49 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:49 - Connection State - Disconnected
2025-06-06 14:25:49 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:49 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:49 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:49 - Connect using port 8883 (TCP)
2025-06-06 14:25:50 - disconnected with result code: 7
2025-06-06 14:25:50 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:50 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:50 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:50 - Connection State - Disconnected
2025-06-06 14:25:50 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:50 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:50 - connected with result code: 0
2025-06-06 14:25:50 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:50 - _on_mqtt_connected called
2025-06-06 14:25:50 - Connection State - Connected
2025-06-06 14:25:50 - Connect using port 8883 (TCP)
2025-06-06 14:25:50 - connected with result code: 0
2025-06-06 14:25:50 - _on_mqtt_connected called
2025-06-06 14:25:50 - Connection State - Connected
2025-06-06 14:25:50 - disconnected with result code: 7
2025-06-06 14:25:50 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:50 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:50 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:50 - Connection State - Disconnected
2025-06-06 14:25:50 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:50 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:50 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:50 - Connect using port 8883 (TCP)
2025-06-06 14:25:51 - connected with result code: 0
2025-06-06 14:25:51 - _on_mqtt_connected called
2025-06-06 14:25:51 - disconnected with result code: 7
2025-06-06 14:25:51 - Connection State - Connected
2025-06-06 14:25:51 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:51 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:51 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:51 - Connection State - Disconnected
2025-06-06 14:25:51 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:51 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:51 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:51 - Connect using port 8883 (TCP)
2025-06-06 14:25:51 - connected with result code: 0
2025-06-06 14:25:51 - _on_mqtt_connected called
2025-06-06 14:25:51 - disconnected with result code: 7
2025-06-06 14:25:51 - Connection State - Connected
2025-06-06 14:25:51 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:51 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:51 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:51 - Connection State - Disconnected
2025-06-06 14:25:51 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:51 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:51 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:51 - Connect using port 8883 (TCP)
2025-06-06 14:25:51 - connected with result code: 0
2025-06-06 14:25:52 - _on_mqtt_connected called
2025-06-06 14:25:52 - Connection State - Connected
2025-06-06 14:25:52 - disconnected with result code: 7
2025-06-06 14:25:52 - Forcing paho disconnect to prevent it from automatically reconnecting
2025-06-06 14:25:52 - MQTTTransportStage: _on_mqtt_disconnect called: The connection was lost.
2025-06-06 14:25:52 - MQTTTransportStage: Unexpected disconnect (no pending connection op)
2025-06-06 14:25:52 - Connection State - Disconnected
2025-06-06 14:25:52 - Cleared all pending method requests due to disconnect
2025-06-06 14:25:52 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:25:52 - ['azure.iot.device.common.transport_exceptions.ConnectionDroppedError: Unexpected disconnection\n']
2025-06-06 14:25:52 - Connect using port 8883 (TCP)
2025-06-06 14:25:52 - message received on $iothub/methods/POST/RunAll/?$rid=1
2025-06-06 14:25:52 - connected with result code: 0
2025-06-06 14:25:52 - _on_mqtt_connected called
2025-06-06 14:25:52 - Connection State - Connected
2025-06-06 14:26:14 - message received on $iothub/methods/POST/StopAll/?$rid=1
2025-06-06 14:26:15 - Sending method response to Hub...
2025-06-06 14:26:15 - publishing on $iothub/methods/res/200/?$rid=1
2025-06-06 14:26:15 - payload published for 3
2025-06-06 14:26:15 - Successfully sent method response to Hub
2025-06-06 14:27:02 - message received on $iothub/methods/POST/StopAll/?$rid=2
2025-06-06 14:27:02 - Bơm 1 đã dừng
2025-06-06 14:27:02 - Bơm 2 đã dừng
2025-06-06 14:27:02 - Bơm 3 đã dừng
2025-06-06 14:27:02 - Bơm 4 đã dừng
2025-06-06 14:27:02 - Bơm 5 đã dừng
2025-06-06 14:27:02 - Bơm 6 đã dừng
2025-06-06 14:27:02 - Bơm 7 đã dừng
2025-06-06 14:27:02 - Bơm 8 đã dừng
2025-06-06 14:27:02 - Bơm 9 đã dừng
2025-06-06 14:27:02 - Tất cả các bơm đã dừng
2025-06-06 14:27:03 - Sending method response to Hub...
2025-06-06 14:27:03 - publishing on $iothub/methods/res/200/?$rid=2
2025-06-06 14:27:03 - payload published for 4
2025-06-06 14:27:03 - Successfully sent method response to Hub
2025-06-06 14:27:35 - Dừng hệ thống
2025-06-06 14:27:36 - Creating client for connecting using MQTT over TCP
2025-06-06 14:27:37 - Connecting to Hub...
2025-06-06 14:27:37 - Connect using port 8883 (TCP)
2025-06-06 14:27:37 - connected with result code: 0
2025-06-06 14:27:37 - _on_mqtt_connected called
2025-06-06 14:27:37 - Connection State - Connected
2025-06-06 14:27:37 - Successfully connected to Hub
2025-06-06 14:27:37 - Enabling feature:methods...
2025-06-06 14:27:37 - subscribing to $iothub/methods/POST/# with qos 1
2025-06-06 14:27:37 - suback received for 1
2025-06-06 14:27:37 - Successfully enabled feature:methods
2025-06-06 14:27:37 - Khởi động hệ thống
2025-06-06 14:28:06 - message received on $iothub/methods/POST/Agitator/?$rid=1
2025-06-06 14:28:06 - Bơm 9 bắt đầu chạy liên tục
2025-06-06 14:28:06 - Máy khấy bắt đầu hoạt động
2025-06-06 14:28:07 - Sending method response to Hub...
2025-06-06 14:28:07 - publishing on $iothub/methods/res/200/?$rid=1
2025-06-06 14:28:07 - payload published for 2
2025-06-06 14:28:07 - Successfully sent method response to Hub
2025-06-06 14:28:15 - message received on $iothub/methods/POST/Agitator/?$rid=2
2025-06-06 14:28:16 - Sending method response to Hub...
2025-06-06 14:28:16 - publishing on $iothub/methods/res/200/?$rid=2
2025-06-06 14:28:16 - payload published for 3
2025-06-06 14:28:16 - Successfully sent method response to Hub
2025-06-06 14:29:09 - Dừng hệ thống
2025-06-06 14:29:11 - Creating client for connecting using MQTT over TCP
2025-06-06 14:29:11 - Connecting to Hub...
2025-06-06 14:29:11 - Connect using port 8883 (TCP)
2025-06-06 14:29:13 - connected with result code: 0
2025-06-06 14:29:13 - _on_mqtt_connected called
2025-06-06 14:29:13 - Connection State - Connected
2025-06-06 14:29:13 - Successfully connected to Hub
2025-06-06 14:29:13 - Enabling feature:methods...
2025-06-06 14:29:13 - subscribing to $iothub/methods/POST/# with qos 1
2025-06-06 14:29:13 - suback received for 1
2025-06-06 14:29:13 - Successfully enabled feature:methods
2025-06-06 14:29:13 - Khởi động hệ thống
2025-06-06 14:29:18 - message received on $iothub/methods/POST/Agitator/?$rid=1
2025-06-06 14:29:18 - Bơm 9 đã dừng
2025-06-06 14:29:18 - Máy khấy đã dừng
2025-06-06 14:29:19 - Sending method response to Hub...
2025-06-06 14:29:19 - publishing on $iothub/methods/res/200/?$rid=1
2025-06-06 14:29:20 - payload published for 2
2025-06-06 14:29:20 - Successfully sent method response to Hub
2025-06-06 14:29:36 - message received on $iothub/methods/POST/Agitator/?$rid=2
2025-06-06 14:29:36 - Bơm 9 bắt đầu chạy liên tục
2025-06-06 14:29:36 - Máy khấy bắt đầu hoạt động
2025-06-06 14:29:37 - Sending method response to Hub...
2025-06-06 14:29:37 - publishing on $iothub/methods/res/200/?$rid=2
2025-06-06 14:29:42 - payload published for 3
2025-06-06 14:29:42 - Successfully sent method response to Hub
2025-06-06 14:32:31 - Dừng hệ thống
2025-06-06 14:32:32 - Creating client for connecting using MQTT over TCP
2025-06-06 14:32:32 - Connecting to Hub...
2025-06-06 14:32:32 - Connect using port 8883 (TCP)
2025-06-06 14:32:32 - connected with result code: 0
2025-06-06 14:32:32 - _on_mqtt_connected called
2025-06-06 14:32:32 - Connection State - Connected
2025-06-06 14:32:32 - Successfully connected to Hub
2025-06-06 14:32:32 - Enabling feature:methods...
2025-06-06 14:32:32 - subscribing to $iothub/methods/POST/# with qos 1
2025-06-06 14:32:32 - suback received for 1
2025-06-06 14:32:32 - Successfully enabled feature:methods
2025-06-06 14:32:32 - Khởi động hệ thống
2025-06-06 14:32:36 - message received on $iothub/methods/POST/Agitator/?$rid=1
2025-06-06 14:32:36 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:32:36 - ['azure.iot.device.iothub.sync_handler_manager.HandlerManagerException: HANDLER (_on_method_request_received): Error during invocation\n']
2025-06-06 14:33:10 - message received on $iothub/methods/POST/Agitator/?$rid=2
2025-06-06 14:33:10 - Exception caught in background thread.  Unable to handle.
2025-06-06 14:33:10 - ['azure.iot.device.iothub.sync_handler_manager.HandlerManagerException: HANDLER (_on_method_request_received): Error during invocation\n']
2025-06-06 14:34:56 - Dừng hệ thống
